### DISCUSSION TITLE
Power efficiency related commands (disable functions during wait periods) using block coding?

### DISCUSSION THREAD

**Discussion by EmperorKabir** [pybricks/support#1982] [Q&A] [answered]
Here is a simple code I am using to push a Lego train around the base of a Christmas tree. It broadcasts from one hub and a receiving hub mimics the actions (for more power) as I am using 2 train motors. It accelerates, hits a top speed, decelerates, hits a minimum speed then looks for a tile using a colour sensor to stop at the station for a while then repeats. (There are a few redundant colour tiles in the setup not being used).

I intend on leaving this running for 15+ hours every day but I suspect that's going to kill my batteries. (I have rechargeables but nevertheless...)

I was wondering if there is anything I can do in block coding to make this more power efficient or if the results would be negligible?
Ideally:
-disable bluetooth broadcasting during the 180,000ms wait (30 minutes) then start it up again when it's time to accelerate.
-disable the color sensor from scanning during the 180,000ms wait then start it up again when it's time to accelerate.
-stop the support hub (code not displayed here) from 'listening' to bluetooth signals during the 180,000ms wait.
-anything else to effectively keep the functionality the same but reduce power consumption

![image](https://github.com/user-attachments/assets/bb091a28-5392-44e5-929b-30e6c7275ebb)

**Comment by laurensvalk** [comment, marked as answer]
On an electromechanical system like this, CPU usuage usually isn't the most costly. You could:

- Stop the motor with coast (instead of duty 0)
- Turn off the hub light (Of course now it looks the same as actually being off. Maybe blink every once in a while to remind you :smile:)
- You can also turn off the color sensor light. Same block as above, but choose the sensor instead of the hub.

**Reply by EmperorKabir** [reply]
Thanks, I've done that and it's working. :) Though it sounds like you believe this will make little difference? Also it sounds like there is no option to disable/re-enable bluetooth broadcasting receiving? (If there was, do you also believe it would make little difference? If you do, is it worth me adding it as an idea request?)

**Reply by laurensvalk** [reply]
You can stop broadcasting by broadcasting the value None. This might save a little bit too.

Yes, turning off as many things as you can that use power (motors, lights, sounds) can make a difference. I just meant that compared to these, computational power is almost negligible, so don't worry too much about program structure and things like that.

**Reply by EmperorKabir** [reply]
Thanks! I tried that but then the receiving hub flashes in 'looking for bluetooth input' mode so I guess I will leave it as continuously broadcasting as at least that way, the lights stay off in the half hour wait which I suspect is more power efficient than having the main hub broadcast 0 but having the support hub continually flashing.

Hopefully I can get a few days at a time but I will see.

Thanks again!

**Reply by dlech** [reply]
CPU and Bluetooth are negligible (microamps) compared to the 50 to 100 milliamps that is required to just have the hub turned on. Lights and motors are the only things that make a measurable difference in power consumption.
